conans/client/generators/cmake_find_package.py

```diff
@@ -43,52 +48,281 @@ class CMakeFindPackageGenerator(Generator):
+
+        ########### FOUND PACKAGE ###################################################################
+        #############################################################################################
+
+        include(FindPackageHandleStandardArgs)
+
+        conan_message(STATUS "Conan: Using autogenerated Find{{ pkg_name }}.cmake")
+        set({{ pkg_name }}_FOUND 1)
+        set({{ pkg_name }}_VERSION "{{ pkg_version }}")
+        set({{ pkg_name }}_COMPONENTS {{ pkg_components }})
+
+        find_package_handle_standard_args({{ pkg_name }} REQUIRED_VARS
+                                          {{ pkg_name }}_VERSION VERSION_VAR {{ pkg_name }}_VERSION)
+        mark_as_advanced({{ pkg_name }}_FOUND {{ pkg_name }}_VERSION)
+
+        if({{ pkg_name }}_FIND_COMPONENTS)
+            foreach(_FIND_COMPONENT {{ '${'+pkg_name+'_FIND_COMPONENTS}' }})
+                list(FIND {{ pkg_name }}_COMPONENTS "{{ pkg_name }}::${_FIND_COMPONENT}" _index)
+                if(${_index} EQUAL -1)
+                    conan_message(FATAL_ERROR "Conan: Component '${_FIND_COMPONENT}' NOT found in package '{{ pkg_name }}'")
+                else()
+                    conan_message(STATUS "Conan: Component '${_FIND_COMPONENT}' found in package '{{ pkg_name }}'")
+                endif()
+            endforeach()
+        endif()
+
+        ########### VARIABLES #######################################################################
+        #############################################################################################
+
+        set({{ pkg_name }}_INCLUDE_DIRS {{ pkg.include_paths }})
+        set({{ pkg_name }}_INCLUDE_DIR {{ pkg.include_path }})
+        set({{ pkg_name }}_INCLUDES {{ pkg.include_paths }})
+        set({{ pkg_name }}_LIB_DIRS {{ pkg.lib_paths }})
+        set({{ pkg_name }}_RES_DIRS {{ pkg.res_paths }})
+        set({{ pkg_name }}_DEFINITIONS {{ pkg.defines }})
+        set({{ pkg_name }}_LINKER_FLAGS_LIST
+                $<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,SHARED_LIBRARY>:{{ pkg.sharedlinkflags_list }}>
+                $<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,MODULE_LIBRARY>:{{ pkg.sharedlinkflags_list }}>
+                $<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,EXECUTABLE>:{{ pkg.exelinkflags_list }}>
+        )
+        set({{ pkg_name }}_COMPILE_DEFINITIONS {{ pkg.compile_definitions }})
+        set({{ pkg_name }}_COMPILE_OPTIONS_LIST "{{ pkg.cxxflags_list }}" "{{ pkg.cflags_list }}")
+        set({{ pkg_name }}_LIBRARIES_TARGETS "") # Will be filled later, if CMake 3
+        set({{ pkg_name }}_LIBRARIES "") # Will be filled later
+        set({{ pkg_name }}_LIBS "") # Same as {{ pkg_name }}_LIBRARIES
+        set({{ pkg_name }}_LIBRARY_LIST {{ pkg.libs }})
+        set({{ pkg_name }}_SYSTEM_LIBS {{ pkg.system_libs }})
+        set({{ pkg_name }}_FRAMEWORK_DIRS {{ pkg.framework_paths }})
+        set({{ pkg_name }}_FRAMEWORKS {{ pkg.frameworks }})
+        set({{ pkg_name }}_FRAMEWORKS_FOUND "") # Will be filled later
```

Comment:
Even if it is ugly, we need to run the `conan_find_apple_frameworks` function to populate this variable. Or append all the frameworks as they are being found for each component.